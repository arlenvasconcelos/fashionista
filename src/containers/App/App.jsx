import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';

import './App.scss';

import Routes from '../../routes';
import {TopBar} from '../../components';
import Cart from '../Cart/Cart';

function App() {
 
  return (
    <div className="app">
      <Router>
        <TopBar />
        <Cart/>
        <Routes/>
      </Router>
    </div>
  );
}

export default App;
