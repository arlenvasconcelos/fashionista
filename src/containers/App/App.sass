@import url('https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap');

*,
*:after,
*:before {
  box-sizing: border-box;
}

html {
  font-size: 14px;
}

body {
  padding-top: 56px;
  font-family: 'Open Sans', sans-serif;
  background-color: #F9F9F9;
}

a {
  text-decoration: none;
  color: #fff;
  font-weight: bold;
}

button {
  background-color: transparent;
  border: none; 
  box-shadow: none;
  outline: none;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.container {
  margin: 0 auto;
  position: relative;
  width: 100%;
}

@media screen and (min-width: 768px) {
  .container {
    width: 668px;
  }
}

@media screen and (min-width: 1024px){
  .container {
    width: 924px;
  }
}