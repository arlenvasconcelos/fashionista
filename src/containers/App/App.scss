
*,
*:after,
*:before {
  box-sizing: border-box;
}

:root{
  --primary-color: #000000;
  --title-color: #000;
  --text-color: #000;
}

html {
  font-size: 14px;
}

body {
  margin: 0;
  font-family: 'Roboto', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  background-color: #F9F9F9;
}

a {
  text-decoration: none;
  color: #000;
}

button {
  background-color: transparent;
  border: none; 
  box-shadow: none;
  outline: none;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
}

.container {
  margin: 0 auto;
}

h1, h2, h3, h4, h5, h6 {
  color: var(--title-color);
  font-family: 'Ubuntu', sans-serif;
}

@media screen and (min-width: 1216px){
  .container {
    width: 1216px;
  }
}