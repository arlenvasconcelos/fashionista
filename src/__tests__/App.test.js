import React from "react";
import renderer from "react-test-renderer";
import App from "../containers/App/App";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("Testing NoProduct", () => {
  const initialState = {
    products: {
      allProducts: [],
      filteredProducts: [],
      openedFilter: false,
    },
    cart: {
      items: [],
      open: false,
    },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
