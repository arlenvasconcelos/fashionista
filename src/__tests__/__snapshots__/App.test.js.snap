// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NoProduct renders correctly 1`] = `
<div
  className="app"
  data-testeid="app"
>
  <header
    className="topbar"
    data-testeid="topbar"
  >
    <div
      className="container"
    >
      <div
        className="topbar__content"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="topbar__logo"
          >
            FASHIONISTA
          </div>
        </a>
        <div
          className="topbar__icons"
        >
          <div
            className="navsearch--ontopbar "
          >
            <div
              className="navsearch__icon"
            >
              <button
                onClick={[Function]}
              >
                <svg
                  fill="none"
                  height={24}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </button>
            </div>
          </div>
          <button
            className="icon"
            onClick={[Function]}
          >
            <svg
              fill="none"
              height={24}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="9"
                cy="21"
                r="1"
              />
              <circle
                cx="20"
                cy="21"
                r="1"
              />
              <path
                d="M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    className="drawer inactive"
  >
    <div
      className="cart"
    >
      <div
        className="cart__header"
      >
        <button
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 5 19 12 12 19"
            />
          </svg>
        </button>
        <span>
          Carrinho (
          0
          )
        </span>
      </div>
      <div
        className="cart__details"
      />
      <div
        className="cart__footer"
      >
        <div
          className="footer__content"
        >
          <p
            className="content__row"
          >
            Total: 
            <span>
              0.00
            </span>
          </p>
          <p
            className="content__row"
          >
            Desconto: 
            <span>
              0.00
            </span>
          </p>
          <p
            className="content__row--total"
          >
            Total com desconto: 
            <span>
              0.00
            </span>
          </p>
        </div>
        <button>
          CONCLUIR COMPRA
        </button>
      </div>
    </div>
  </div>
  <div
    className="loading"
  />
  <footer
    className="footer"
    style={
      Object {
        "position": "fixed",
      }
    }
  >
    <div
      className="container"
    >
      <h3>
        Fashionista E-commerce
      </h3>
      <p>
        Acompanhe-nos tamb√©m nas redes sociais
      </p>
      <div
        className="footer__icons"
      >
        <a
          href="#"
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="20"
              rx="5"
              ry="5"
              width="20"
              x="2"
              y="2"
            />
            <path
              d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
            />
            <line
              x1="17.5"
              x2="17.51"
              y1="6.5"
              y2="6.5"
            />
          </svg>
        </a>
        <a
          href="#"
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
            />
          </svg>
        </a>
        <a
          href="#"
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z"
            />
          </svg>
        </a>
      </div>
    </div>
  </footer>
</div>
`;

exports[`With React Testing Library renders correctly 1`] = `
<div
  className="app"
  data-testeid="app"
>
  <header
    className="topbar"
    data-testeid="topbar"
  >
    <div
      className="container"
    >
      <div
        className="topbar__content"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="topbar__logo"
          >
            FASHIONISTA
          </div>
        </a>
        <div
          className="topbar__icons"
        >
          <div
            className="navsearch--ontopbar "
          >
            <div
              className="navsearch__icon"
            >
              <button
                onClick={[Function]}
              >
                <svg
                  fill="none"
                  height={24}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </button>
            </div>
          </div>
          <button
            className="icon"
            onClick={[Function]}
          >
            <svg
              fill="none"
              height={24}
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="9"
                cy="21"
                r="1"
              />
              <circle
                cx="20"
                cy="21"
                r="1"
              />
              <path
                d="M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    className="drawer inactive"
  >
    <div
      className="cart"
    >
      <div
        className="cart__header"
      >
        <button
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 5 19 12 12 19"
            />
          </svg>
        </button>
        <span>
          Carrinho (
          0
          )
        </span>
      </div>
      <div
        className="cart__details"
      />
      <div
        className="cart__footer"
      >
        <div
          className="footer__content"
        >
          <p
            className="content__row"
          >
            Total: 
            <span>
              0.00
            </span>
          </p>
          <p
            className="content__row"
          >
            Desconto: 
            <span>
              0.00
            </span>
          </p>
          <p
            className="content__row--total"
          >
            Total com desconto: 
            <span>
              0.00
            </span>
          </p>
        </div>
        <button>
          CONCLUIR COMPRA
        </button>
      </div>
    </div>
  </div>
  <div
    className="loading"
  />
  <footer
    className="footer"
    style={
      Object {
        "position": "fixed",
      }
    }
  >
    <div
      className="container"
    >
      <h3>
        Fashionista E-commerce
      </h3>
      <p>
        Acompanhe-nos tamb√©m nas redes sociais
      </p>
      <div
        className="footer__icons"
      >
        <a
          href="#"
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="20"
              rx="5"
              ry="5"
              width="20"
              x="2"
              y="2"
            />
            <path
              d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
            />
            <line
              x1="17.5"
              x2="17.51"
              y1="6.5"
              y2="6.5"
            />
          </svg>
        </a>
        <a
          href="#"
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
            />
          </svg>
        </a>
        <a
          href="#"
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z"
            />
          </svg>
        </a>
      </div>
    </div>
  </footer>
</div>
`;
